From b3093323fdc4e436976c97bb3b1e50b78c3d1fc9 Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Tue, 13 Dec 2022 15:41:44 +0800
Subject: [PATCH] gstreamer1.0-plugins-bad: CB1 multiqueue handle event
 AML-SET-MAX-BYTE-SIZE  [1/1]

PD#SWPL-104970

Problem:
ts video data is more than setted multiqueue buffer size
at the beginning which will stuck the pipeline when multiqueue is filled

Solution:
tsdemux seend a event to multiqueue
when multiqueue get this event, set the buffer size to default 10M.

Verify:
AH212
---
 plugins/elements/gstmultiqueue.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/plugins/elements/gstmultiqueue.c b/plugins/elements/gstmultiqueue.c
index d4feb89..bcb55c5 100644
--- a/plugins/elements/gstmultiqueue.c
+++ b/plugins/elements/gstmultiqueue.c
@@ -1672,6 +1672,20 @@ gst_single_queue_push_one (GstMultiQueue * mq, GstSingleQueue * sq,
         /* Applying the gap may have made the queue non-full again, unblock it if needed */
         gst_data_queue_limits_changed (sq->queue);
         break;
+      case GST_EVENT_CUSTOM_DOWNSTREAM_STICKY:
+        if(gst_event_has_name(event, "AML-SET-MAX-BYTE-SIZE")) {
+          GST_DEBUG_OBJECT (mq, "Handle event AML-SET-MAX-BYTE-SIZE");
+          GST_MULTI_QUEUE_MUTEX_LOCK (mq);
+          mq->max_size.bytes = DEFAULT_MAX_SIZE_BYTES;
+          SET_CHILD_PROPERTY (mq, bytes);
+          mq->max_size.time = DEFAULT_MAX_SIZE_TIME * 4;
+          SET_CHILD_PROPERTY (mq, time);
+          GST_MULTI_QUEUE_MUTEX_UNLOCK (mq);
+          gst_multi_queue_post_buffering (mq);
+          gst_event_unref(event);
+          result = GST_FLOW_OK;
+        }
+        break;
       default:
         break;
     }
-- 
2.25.1

